DllImportGenerator/DllImportGenerator/DllImportGenerator.cs

```diff
@@ -79,83 +81,36 @@ public void Initialize(InitializationContext context)
             DllImportStub stub,
             AttributeSyntax dllImportAttr)
         {
-            const string SingleDepth = "    ";
-            var currentIndent = string.Empty;
-
-            // Declare namespace
-            if (!(stub.StubTypeNamespace is null))
-            {
-                builder.AppendLine($@"namespace {stub.StubTypeNamespace}
-{{");
-                currentIndent += SingleDepth;
-            }
-
-            // Print type declarations
-            var typeIndentStack = new Stack<string>();
-            foreach (var typeDecl in stub.StubContainingTypesDecl)
-            {
-                builder.AppendLine($@"{currentIndent}{typeDecl}
-{currentIndent}{{");
-
-                typeIndentStack.Push(currentIndent);
-                currentIndent += SingleDepth;
-            }
-
-            // Begin declare function
-            builder.Append(
-$@"{currentIndent}{userDeclaredMethod.Modifiers} {stub.StubReturnType} {userDeclaredMethod.Identifier}(");
-
-            char delim = ' ';
-            foreach (var param in stub.StubParameters)
-            {
-                builder.Append($"{delim}{param.Type} {param.Name}");
-                delim = ',';
-            }
-
-            // End declare function
-            builder.AppendLine(
-$@")
-{currentIndent}{{");
-
-            // Insert lines into function
-            foreach (var line in stub.StubCode)
-            {
-                builder.AppendLine($@"{currentIndent}{SingleDepth}{line}");
-            }
-
-            builder.AppendLine(
-$@"{ currentIndent}}}
-
-{currentIndent}[{dllImportAttr}]");
+            // Create stub function
+            var stubMethod = MethodDeclaration(stub.StubReturnType, userDeclaredMethod.Identifier)
+                .WithModifiers(userDeclaredMethod.Modifiers)
+                .WithParameterList(ParameterList(SeparatedList(stub.StubParameters)))
+                .WithBody(stub.StubCode);
 
             // Create the DllImport declaration.
-            builder.Append($"{currentIndent}extern private static {stub.DllImportReturnType} {stub.DllImportMethodName}");
-            if (!stub.DllImportParameters.Any())
+            var dllImport = stub.DllImportDeclaration.AddAttributeLists(
+                AttributeList(
+                    SingletonSeparatedList<AttributeSyntax>(dllImportAttr)));
+
+            // Add stub function and DllImport declaration to the containing types
+            MemberDeclarationSyntax containingType = stub.StubContainingTypes.First()
+                .AddMembers(stubMethod, dllImport );
+            foreach (var typeDecl in stub.StubContainingTypes.Skip(1))
```

Comment:
Comment on why `Skip(1)`. I assume we are skipping the return?

Reply:
`containingType` is initialized using the first one above. Will comment about skipping and assert that `StubContainingTypes` should not be empty.